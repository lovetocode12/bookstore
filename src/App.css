.app-container {
  display: grid;
  grid-template-rows: 10vh 90vh;
}


.header {
  display: grid;
  grid-template-columns: 1fr 2fr;
  background: #eee;
}

.header-title{
  display: flex;
  justify-content: flex-start;
  align-items: center;
  color: #f07e9a;
  margin-left: 1em;
}

.nav-container{
  display: flex;
  justify-content: center;
  align-items: center;
}

.nav{
  display: flex;
  width: 100%;
  justify-content: flex-end;
  align-items: center;
}

.nav-list{
  list-style: none;
  width: 100%;
  display: flex;
  justify-content: flex-end;
  align-items: center;
}

.nav-item{
  display: flex;
  flex-basis: 201px;
  justify-content: center;
  align-items: center;
}

.nav-item:nth-child(2){
  border-left: 2px solid rgb(211, 201, 201);
  border-right: 2px solid rgb(211, 201, 201);
}

h1,h2,h3,h4,h5,h6{
  margin: 0;
  display: inline;
}

a{
  text-decoration: none;
}


@media only screen and (max-width : 720px){

  .app-container {
    grid-template-rows: 20vh 80vh;
  }

  .header {
  grid-template-columns: 1fr;
    grid-template-rows: 1fr 1fr;
  }
}

@media only screen and (max-width : 360px){

  .app-container {
    grid-template-rows: 30vh 1fr;
  }

  .header {
  grid-template-columns: 1fr;
    grid-template-rows: 10vh 1fr;
  }

  .nav-item {
    flex-basis: 30px;
  }

  .nav-list{
    flex-direction: column;
  }

  .nav-item:nth-child(2){
    border-top: 2px solid rgb(211, 201, 201);
    border-bottom: 2px solid rgb(211, 201, 201);
    border-left:none;
    border-right: none;
  }
}

