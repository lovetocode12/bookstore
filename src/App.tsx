import React from 'react';
import './App.css';
import configureStore from './store'
import { Provider } from 'react-redux'
import BookList from './components/BookList/BookList';
import { BrowserRouter as Router, Route, Switch } from "react-router-dom";
import BookDetails from './components/BookDetails/BookDetails';
import ErrorBoundary from './common/ErrorBoundary';
import { PersistGate } from 'redux-persist/integration/react'
import OrdersList from './components/OrdersList/OrdersList';
import Cart from './components/Cart/Cart';
import NavBar from './components/Nav/Nav';
import TitleProvider from './common/TitleProvider';

type Props = {};

const App: React.FC<Props> = () => {

  const { store, persistor } = configureStore();

  return (
    <Provider store={store}>
      <PersistGate loading={null} persistor={persistor}>
        <TitleProvider>
          <Router>
            <div className="app-container">
              <header className="header">
                <NavBar></NavBar>
              </header>
              <main className="main">
                <ErrorBoundary>
                  <Switch>
                    <Route exact path="/" component={BookList}></Route>
                    <Route exact path="/details/:book_id" component={BookDetails}></Route>
                    <Route exact path="/orders" component={OrdersList}></Route>
                    <Route exact path="/cart" component={Cart}></Route>
                  </Switch>
                </ErrorBoundary>
              </main>
            </div>
          </Router>
        </TitleProvider>
      </PersistGate>
    </Provider >
  );
}

export default App;
