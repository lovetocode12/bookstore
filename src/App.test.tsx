import React from 'react';
import App from './App';
import { shallow } from '../enzyme';

describe('App Component', () => {

  it('should render correctly', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should have 4 routes', () => {
    const wrapper = shallow(<App/>);
    expect(wrapper.find('Route').length).toEqual(4);
  });

})

