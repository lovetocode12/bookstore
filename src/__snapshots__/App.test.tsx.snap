// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component should render correctly 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <PersistGate
    loading={null}
    persistor={
      Object {
        "dispatch": [Function],
        "flush": [Function],
        "getState": [Function],
        "pause": [Function],
        "persist": [Function],
        "purge": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <TitleProvider>
      <BrowserRouter>
        <div
          className="app-container"
        >
          <header
            className="header"
          >
            <Nav />
          </header>
          <main
            className="main"
          >
            <ErrorBoundary>
              <Switch>
                <Route
                  component={[Function]}
                  exact={true}
                  path="/"
                />
                <Route
                  component={[Function]}
                  exact={true}
                  path="/details/:book_id"
                />
                <Route
                  component={[Function]}
                  exact={true}
                  path="/orders"
                />
                <Route
                  component={[Function]}
                  exact={true}
                  path="/cart"
                />
              </Switch>
            </ErrorBoundary>
          </main>
        </div>
      </BrowserRouter>
    </TitleProvider>
  </PersistGate>
</Provider>
`;
